[Draft] Upgrade Bootstrap 3.3.7 to 5.3.3

sahil-idk opened the pull request:
## Overview
This PR upgrades Bootstrap from 3.3.7 to 5.3.3 as requested in #69.  @AyushDharDubey 

## Progress
-  Bootstrap 5.3.3 files added to `static_src/lib/`
-  Navbar migrated to Bootstrap 5 structure
-  Custom gradient styling added to `main.css`
-  Remaining templates to be migrated (panels → cards, etc.)

## Approach
Following maintainer guidance:
- Bootstrap 5 files placed in `mainapp/static_src/lib/`
- Build process (`npm run build`) copies to `mainapp/static/mainapp/lib/`
- No CDN dependencies

## Testing
-  Navbar displays correctly with dark gradient
-  Responsive collapse works on mobile
-  No console errors

## Next Steps
Planning to migrate in this order:
1. `modelpanel.html` (card component used on 3 pages)
2. `docs.html` (API documentation panels)
3. `model.html` (model detail page)
4. `modelform.html` (form styling)
5. Other template files

## Questions
- Should I maintain the exact visual appearance, or is modernization OK?
- Any specific components/pages that are priority?

Closes #69

sahil-idk committed:
Add Bootstrap migration tracking file

sahil-idk committed:
Update layout.html for Bootstrap 5.3.3

- Removed bootstrap-theme.min.css reference (not in BS5)
- Updated comment to indicate Bootstrap 5.3.3 bundle
- Bootstrap assets replaced locally (not in git - build artifacts)
- Updated package-lock.json from npm build
- Site currently broken - templates not yet migrated

sahil-idk committed:
Migrate navbar to Bootstrap 5

- Replaced Bootstrap 3 navbar with Bootstrap 5 version
- navbar-inverse → navbar-dark bg-dark
- navbar-toggle → navbar-toggler
- data-toggle → data-bs-toggle
- data-target → data-bs-target
- Added nav-item and nav-link classes
- navbar-right → ms-auto
- Fixed footer class: navbar-fixed-bottom → fixed-bottom
- Bootstrap 5 assets updated locally (not in git)

Navbar now displays correctly with dark styling and proper responsive behavior

sahil-idk committed:
Add Bootstrap 5 navbar gradient and responsive styles

- Added gradient background for navbar (replaces bootstrap-theme)
- Added smooth collapse animation
- Updated search bar responsive widths
- Added focus styles for form controls

sahil-idk committed:
Upgrade to Bootstrap 5.3.3

- Added Bootstrap 5.3.3 files to static_src/lib/
- Updated navbar to Bootstrap 5 structure
- Migrated navbar classes: navbar-inverse → navbar-dark bg-dark
- Updated navbar toggle and collapse attributes for BS5
- Added navbar gradient styling to main.css
- Fixed footer class: navbar-fixed-bottom → fixed-bottom

Resolves #69

AyushDharDubey commented:
> Should I maintain the exact visual appearance, or is modernization OK?

As starters, it would be best to maintain the design as far as possible for the sake of easier transition, once we are on board Bootstrap 5 completely, then we may consider further modernization.

> Any specific components/pages that are priority?

None as such, the changes should be applicable on the whole project

sahil-idk commented:
<img width="1919" height="908" alt="image" src="https://github.com/user-attachments/assets/ce99ce60-72ea-4017-8c1f-b5873996195e" />

## Current Progress

The navbar has been successfully migrated to Bootstrap 5.3.3 (screenshot above). Here's the current state:



### 🤝 Request for Review
@AyushDharDubey Could you review the approach taken so far? I'll be committing the remaining migrations incrementally. If you spot any discrepancies or have suggestions, please let me know early so I can adjust the pattern before completing all templates.

This will help avoid any large-scale refactoring later. Thanks!

AyushDharDubey commented on mainapp/static_src/public/mainapp/main.css:
```diff
@@ -208,4 +208,73 @@ form > #map-descriptor {
     color: rgba(255, 255, 255, 0.75);
-}
\ No newline at end of file
+}
```
Is the piece llm generated?

It won't serve the purpose if we are effectively cluttering our stylesheets with custom styles for every small bit.

Looks like you weren't able to load bootstrap and ultimately fell back to custom CSS styling

sahil-idk replied:
yeah i checked this with cdn and i think it was hover properties  and custom paddings in align left were not working as some inbuilt old bootstrap classes like navbar-form and navbar-right have been depreceated

AyushDharDubey reviewed (changes requested):
try getting to load the bootstrap5 version first instead of the bootstrap3
you might want to inspect the page to check if you are loading the correct assets

sahil-idk commented:
> > now it loads perfectly
> 
> That's strange, as with the current state, it should still not load: the url of bootstrap5 is at `mainapp/static_src/lib/bootstrap.min.css` and the html template expects it at `mainapp/static_src/mainapp/lib/bootstrap.min.css`

<img width="1567" height="378" alt="image" src="https://github.com/user-attachments/assets/8a7d785a-2d11-48b2-a753-8badfafc181e" />

Hmm wierd , then i cant understand it loads here on networks and how i reference is
 `<link rel="stylesheet" href="{% static 'mainapp/lib/bootstrap.min.css' %}">
`

like how it was for bs3

sahil-idk committed:
Place Bootstrap 5.3.3 in correct source location

- Added Bootstrap 5.3.3 files to static_src/public/mainapp/lib/ (source)
- Removed bootstrap-theme.min.css (not used in Bootstrap 5)
- Updated layout.html to use local Bootstrap files from lib/
- Files placed in public/ directory so Vite copies them during build
- bootstrap.min.css: 232KB (v5.3.3)
- bootstrap.min.js: 80KB (v5.3.3 bundle with Popper)

Follows maintainer guidance: Vite copies from static_src/public/ → static/ during npm run build

sahil-idk commented:
> > and about changing the location of bs5 packages i will do that its correctly placed in the static 3dmr\mainapp\static\mainapp\lib\bootstrap.min.css but something happened in build that has messed it up i suppose
> 
> I see the real culprit now, django fetches statics from `3dmr\mainapp\static` but it isn't meant to be git tracked, that's the reason it was working as expected at your local but the remote was wary. You should really avoid tampering the `3dmr\mainapp\static\`, instead, after each change in `3dmr\mainapp\static_src` hit `npm run build`.

Yeah i also figured that one last night and the latest commit i fixed that , made changes to static src and ran npm run build the folder was  mainapp\static_src\public\mainapp\lib  where i added the new bs5 and ran build and removed the bs3 ones you can check here the changes  https://github.com/fossgis/3dmr/pull/74/commits/f082dfe9b34c34f9dcdd95d14e72719810e945bf  and yeah will stick to custom changes  will push more migrations meanwhile

sahil-idk committed:
Migrate modelpanel and index to Bootstrap 5

- Update modelpanel.html: panel → card structure
- Replace .label with .badge classes
- Update pagination to Bootstrap 5 structure
- Use pure Bootstrap 5 utilities (no custom CSS)
- Add bg-light, bg-secondary, text-uppercase classes
- Remove btn-group, replace with proper pagination

sahil-idk committed:
Migrate search.html to Bootstrap 5

- Replace page-header with h1 and Bootstrap 5 utilities
- Use text-muted and fs-3 for search parameters styling
- Migrate pagination from btn-group to proper pagination component
- Add horizontal rule below heading to match design
- Remove Bootstrap 3 classes: page-header, btn-default, btn-group

sahil-idk committed:
Migrate docs.html to Bootstrap 5

- Convert panel structure to card components
- Replace label badges with Bootstrap 5 badge classes
- Add gray background containers for all code blocks
- Update form elements (textarea, submit button)
- Fix CSS target selector for card highlighting
- Add horizontal rule below main heading
- Use h4 class for smaller sidebar headings

Related to #69

sahil-idk committed:
model migration

sahil-idk commented:
@AyushDharDubey   can yo review the progress please

AyushDharDubey commented:
I'll have to close this.
The discussion increasingly suggests I am talking to an LLM, which is not sustainable for work of this scope.

sahil-idk commented:
> I'll have to close this. The discussion increasingly suggests I am talking to an LLM, which is not sustainable for work of this scope.

i have just used the llm to generate the progress  comment but you can see my commits are not llm and my own changes , can we not find a workaround this so that work is continued , used llm to generate comments so that it stay organised but code is not llm generated will that be a problem @AyushDharDubey  @tordanik
